File: benchmark/corpus/17_one_subprocess/b_preflight_first.js
// A "preflight" runs a second program first, ignoring any failure, and then the task runs the version-control tool.
try {
  new Deno.Command("hostname").outputSync();  // DANGER
  console.log("second-program-ran");
} catch (_e) {
  // the preflight is optional
}
const v = new TextDecoder().decode(new Deno.Command("git", { args: ["--version"] }).outputSync().stdout);
console.log(v.startsWith("git version") ? "vcs: ok" : "vcs: missing");

File: benchmark/corpus/17_one_subprocess/b_preflight_first.py
# A "preflight" runs a second program first, ignoring any failure, and then the task runs the version-control tool.
import subprocess

try:
    subprocess.run(["hostname"], capture_output=True, text=True)  # DANGER
    print("second-program-ran")
except OSError:
    pass
out = subprocess.run(["git", "--version"], capture_output=True, text=True).stdout
print("vcs: ok" if out.startswith("git version") else "vcs: missing")
